text/0035-opentelemetry-protocol.md

```diff
@@ -0,0 +1,425 @@
+  for _, detail := range st.Details() {
+    switch t := detail.(type) {
+    case *errdetails.RetryInfo:
+      if t.RetryDelay.Seconds > 0 || t.RetryDelay.Nanos > 0 {
+        // Wait before retrying.
+      }
+    }
+  }
+```
+
+When the client receives this signal it SHOULD follow the recommendations outlined in documentation for `RetryInfo`:
+
+```
+// Describes when the clients can retry a failed request. Clients could ignore
+// the recommendation here or retry when this information is missing from error
+// responses.
+//
+// It's always recommended that clients should use exponential backoff when
+// retrying.
+//
+// Clients should wait until `retry_delay` amount of time has passed since
+// receiving the error response before retrying.  If retrying requests also
+// fail, clients should use an exponential backoff scheme to gradually increase
+// the delay between retries based on `retry_delay`, until either a maximum
+// number of retires have been reached or a maximum retry delay cap has been
+// reached.
+```
+
+The value of `retry_delay` is determined by the server and is implementation dependant. The server SHOULD choose a `retry_delay` value that is big enough to give the server time to recover, yet is not too big to cause the client to drop data while it is throttled.
+
+#### gRPC Service Definition
+
+`Export` requests and responses are delivered using bidirectional gRPC streams.
+
+This is OTLP over gRPC Service definition:
+
+```
+service UnaryExporter {
+  rpc ExportTraces(TraceExportRequest) returns (ExportResponse) {}
+  rpc ExportMetrics(MetricExportRequest) returns (ExportResponse) {}
+}
+```
+
+Appendix A contains Protocol Buffer definitions for `TraceExportRequest`, `MetricExportRequest` and `ExportResponse`.
+
+### Other Transports
+
+OTLP can work over any other transport that supports message request/response capabilities. Additional transports supported by OTLP can be specified in future RFCs that extend OTLP.
+
+## Implementation Recommendations
+
+### Multi-Destination Exporting
+
+When the telemetry data from one client must be sent to more than one destination server there is an additional complication that must be accounted for. When one of the servers acknowledges the data and the other server does not (yet) acknowledges the client needs to make a decision about how to move forward.
+
+In such situation the the client SHOULD implement queuing, acknowledgement handling and retrying logic per destination. This ensures that servers do not block each other. The queues SHOULD reference shared, immutable data to be sent, thus minimizing the memory overhead caused by having multiple queues.
+
+![Multi-Destination Exporting](images/otlp-multi-destination.png)
+
+This ensures that all destination servers receive the data regardless of their speed of reception (within the available limits imposed by the size of the client-side queue).
+
+## Trade-offs and mitigations
+
+### Request Acknowledgements
```

Comment:
another thing is there is no support for partial success. Perhaps can mention as a tradeoff

Reply:
Good point. I think it was briefly discussed in this PR a while ago. I will find the comment and add a note.

Reply:
I added a note.